src/internal/m365/collection/site/restore_test.go

```diff
@@ -234,46 +478,40 @@ func deleteList(
 	}
 }
 
-func setupDependencies(
-	suite tester.Suite,
-	ac api.Client,
-	siteID string,
-	creds account.M365Config,
-	listTemplate string) (
-	string, listsRestoreHandler, string, *dataMock.Item,
-) {
-	t := suite.T()
-	testName := "MockListing"
-
-	lrh := NewListsRestoreHandler(siteID, ac.Lists())
-
-	service := createTestService(t, creds)
-
-	listInfo := models.NewListInfo()
-	listInfo.SetTemplate(ptr.To(listTemplate))
-
-	listing := spMock.ListDefault("Mock List")
-	listing.SetDisplayName(&testName)
-	listing.SetList(listInfo)
+func generateListData(
+	t *testing.T,
+	service *graph.Service,
+	list models.Listable,
+) *dataMock.Item {
+	listName := ptr.Val(list.GetDisplayName())
 
-	byteArray, err := service.Serialize(listing)
+	byteArray, err := service.Serialize(list)
 	require.NoError(t, err, clues.ToCore(err))
 
-	destName := testdata.DefaultRestoreConfig("").Location
-
 	listData, err := data.NewPrefetchedItemWithInfo(
 		io.NopCloser(bytes.NewReader(byteArray)),
-		testName,
-		details.ItemInfo{SharePoint: api.ListToSPInfo(listing)})
+		listName,
+		details.ItemInfo{SharePoint: api.ListToSPInfo(list)})
 	require.NoError(t, err, clues.ToCore(err))
 
 	r, err := readers.NewVersionedRestoreReader(listData.ToReader())
 	require.NoError(t, err)
 
 	mockData := &dataMock.Item{
-		ItemID: testName,
+		ItemID: listName,
 		Reader: r,
 	}
 
-	return testName, lrh, destName, mockData
+	return mockData
+}
+
+func createList(listTemplate, listDisplayName string) models.Listable {
```

Comment:
nit: prefer `stub` to suggest we're mocking out a list.  `create` could be confused with the process of adding it to an external dataset.
```suggestion
func stubList(listTemplate, listDisplayName string) models.Listable {
```